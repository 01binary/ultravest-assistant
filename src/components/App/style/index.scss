@import 'fonts';
@import 'colors';
@import 'layout';

// Default or Mobile
html,
body {
	font-family: $font-primary;
	font-weight: $font-weight-primary;
	font-size: $font-size-text;
	line-height: $spacing-single;

	color: $color-text-primary;
	background-color: $color-background-primary;

	// Variable width with no gutter
	margin: 0;
}

// Desktop or Tablet
@media (min-width: $spacing-max) {
	body {
		// Fixed width with auto gutter
		margin: 0 auto;
		width: $spacing-max;
	}
}

h1 {
	font-size: $font-size-title;
}

h2 {
	font-size: $font-size-heading;
}

h3 {
	font-size: $font-size-text;
}

h1, h2, h3 {
	margin-top: 0;
	margin-bottom: $spacing-single;
}

article {
	padding: $spacing-single;
	border-bottom: $border dashed $color-border;
}

#app {
	display: flex;
	flex-flow: row wrap;
}
