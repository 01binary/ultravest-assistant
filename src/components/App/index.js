
import { h } from 'preact';
import Header from '../Header';
import Flask from '../Flask';
import Investment from '../Investment';
import Burnout from '../Burnout';
import Footer from '../Footer';
import getFlask from '../../selectors/getFlask';
import style from './style';

/**
 * Ultravest Assistant App
 * @param {object} props - The composed application state.
 * @returns {function} - A stateless component.
 */
const App = props => (
	<form id="app" class={style.app}>
		<Header />

		<main>
			<Flask {...getFlask(props)} />
			<Investment {...props} />
			<Burnout {...props} />
		</main>

		<Footer />
	</form>
);

export default App;
