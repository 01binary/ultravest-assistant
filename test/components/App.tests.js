import { h, render } from 'preact';
import { expect } from 'chai';
import withState from '../../src/composers';
import App from '../../src/components/App';

describe('App', () => {

	let wrapper;

	before(() => {
		wrapper = render(withState(<App />), global.document);
		console.log(wrapper);
	});

	after(() => {
		wrapper = null;
	});

	it('should render wrapper', () => {
		
		expect(wrapper.nodeName).to.equal('DIV');
		expect(wrapper.attributes).to.contain({
			name: 'id',
			value: 'app'
		});
	});

	it('should render Flask');

	it('should render Investment');

	it('should render Burnout');

	it('should render Result');

	it('should render Summary');
});
