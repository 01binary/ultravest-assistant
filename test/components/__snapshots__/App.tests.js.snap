// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
preact-render-spy (1 nodes)
-------
<form
  id="app"
  class="app"
  method="post"
  action="/"
>
  <header>
    <h1>
      UltraVest 
      <span>assistant</span>
    </h1>
    <p>
      <strong>Calculate</strong>
       materials and burnout program for your project.
    </p>
    <p>
      Bookmark or 
      <strong>share</strong>
       via permalink.
    </p>
    <input
      type="submit"
      name="action"
      value="Share"
     />
    </input>
    <a
      href="https://github.com/01binary/ultravest-assistant/"
      target="_blank"
      rel="noopener noreferrer"
    >
      Fork
    </a>
  </header>
  <main>
    <article class="flask">
      <h2>flask</h2>
      <label for="flask-preset">
        preset
        <select
          name="flask-preset"
          value="3.5 × 10"
          onChange={[Function anonymous]}
        >
          <option selected={false}>3.5 × 5</option>
          <option selected={true}>3.5 × 10</option>
          <option selected={false}>4 × 10</option>
          <option selected={false}>4 × 14</option>
          <option selected={false}>custom</option>
        </select>
        <input
          type="submit"
          name="action"
          value="Add flask preset"
          onClick={[Function anonymous]}
         />
        </input>
        <input
          type="submit"
          name="action"
          value="Remove flask preset"
          onClick={[Function anonymous]}
         />
        </input>
      </label>
      <label for="flask-diameter">
        diameter
        <input
          name="flask-diameter"
          type="number"
          step="0.1"
          value={3.5}
          onChange={[Function anonymous]}
         />
        </input>
      </label>
      <label for="flask-height">
        height
        <input
          name="flask-height"
          type="number"
          step="0.1"
          value={10}
          onChange={[Function anonymous]}
         />
        </input>
      </label>
    </article>
    <article class="investment">
      <h2>investment</h2>
      <label for="investment-ratio">ratio</label>
      <select
        name="investment-ratio"
        value="40/100"
        onChange={[Function anonymous]}
      >
        <option selected={false}>38/100</option>
        <option selected={false}>39/100</option>
        <option selected={true}>40/100</option>
        <option selected={false}>41/100</option>
        <option selected={false}>42/100</option>
      </select>
      <output>
        <dl>
          <dt>investment</dt>
          <dd>1963</dd>
        </dl>
        <dl>
          <dt>water</dt>
          <dd>789</dd>
        </dl>
      </output>
    </article>
    <article class="burnout">
      <h2>burnout</h2>
      <output>
        <dl>
          <dt>preset</dt>
          <dd>
            4 X 8 in
            <span class="longest">longest</span>
          </dd>
          <dt>time</dt>
          <dd>15 hours</dd>
        </dl>
      </output>
      <section class="diagram">
        <a href="#">
          ←
          <span>Previous Step</span>
        </a>
        <a href="#">
          <span>Next Step</span>
          →
        </a>
        <article class="segment raise">
          <figure class="temperature">
            <svg></svg>
            <figcaption>
              <h3>temp</h3>
              ambient
            </figcaption>
          </figure>
          <figure class="rate">
            <svg></svg>
            <figcaption>
              <h3>rate</h3>
              300°
              <span class="units">F/hr</span>
            </figcaption>
          </figure>
          <figure class="time">
            <svg></svg>
            <figcaption>
              <h3>hold</h3>
              3
              <span class="units">hours</span>
            </figcaption>
          </figure>
        </article>
        <article class="segment raise">
          <figure class="temperature">
            <svg></svg>
            <figcaption>
              <h3>temp</h3>
              700°
              <span class="units">F</span>
            </figcaption>
          </figure>
          <figure class="rate">
            <svg></svg>
            <figcaption>
              <h3>rate</h3>
              200°
              <span class="units">F/hr</span>
            </figcaption>
          </figure>
          <figure class="time">
            <svg></svg>
            <figcaption>
              <h3>hold</h3>
              2
              <span class="units">hours</span>
            </figcaption>
          </figure>
        </article>
        <article class="segment raise">
          <figure class="temperature">
            <svg></svg>
            <figcaption>
              <h3>temp</h3>
              1350°
              <span class="units">F</span>
            </figcaption>
          </figure>
          <figure class="rate">
            <svg></svg>
            <figcaption>
              <h3>rate</h3>
              217°
              <span class="units">F/hr</span>
            </figcaption>
          </figure>
          <figure class="time">
            <svg></svg>
            <figcaption>
              <h3>hold</h3>
              3
              <span class="units">hours</span>
            </figcaption>
          </figure>
        </article>
        <article class="segment lower">
          <figure class="temperature">
            <svg></svg>
            <figcaption>
              <h3>temp</h3>
              700°
              <span class="units">F</span>
            </figcaption>
          </figure>
          <figure class="rate">
            <svg></svg>
            <figcaption>
              <h3>rate</h3>
              700°
              <span class="units">F/hr</span>
            </figcaption>
          </figure>
          <figure class="time">
            <svg></svg>
            <figcaption>
              <h3>hold</h3>
              2
              <span class="units">hours</span>
            </figcaption>
          </figure>
        </article>
      </section>
      <section class="steps">
        <input
          type="radio"
          name="program-view"
          id="steps"
          checked={true}
          onChange={[Function anonymous]}
         />
        </input>
        <label for="steps">steps</label>
        <input
          type="radio"
          name="program-view"
          id="segments"
          checked={false}
          onChange={[Function anonymous]}
         />
        </input>
        <label for="segments">segments</label>
        <table>
          <thead>
            <tr>
              <th>segment</th>
              <th>rate</th>
              <th>temp</th>
              <th>hold</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>water removal</td>
              <td>300</td>
              <td>300</td>
              <td>3</td>
            </tr>
            <tr>
              <td>thermal transition</td>
              <td>200</td>
              <td>700</td>
              <td>2</td>
            </tr>
            <tr>
              <td>pattern removal</td>
              <td>217</td>
              <td>1350</td>
              <td>3</td>
            </tr>
            <tr>
              <td>casting</td>
              <td>700</td>
              <td>700</td>
              <td>2</td>
            </tr>
          </tbody>
        </table>
        <ol>
          <li>
            raise to
            <span> 300</span>
            ° over 
            <span> 1 </span>
            hour
          </li>
          <li>
            hold
            <span> 300</span>
            ° for 
            <span> 3 </span>
            hours
          </li>
          <li>
            raise to
            <span> 700</span>
            ° over 
            <span> 2 </span>
            hours
          </li>
          <li>
            hold
            <span> 700</span>
            ° for 
            <span> 2 </span>
            hours
          </li>
          <li>
            raise to
            <span> 1350</span>
            ° over 
            <span> 3 </span>
            hours
          </li>
          <li>
            hold
            <span> 1350</span>
            ° for 
            <span> 3 </span>
            hours
          </li>
          <li>
            lower to
            <span> 700</span>
            ° over 
            <span> 0.9 </span>
            hour
          </li>
          <li>
            hold
            <span> 700</span>
            ° for 
            <span> 2 </span>
            hours
          </li>
        </ol>
      </section>
    </article>
  </main>
</form>

`;
